### PR TITLE
TEZ-4757: APPROXIMATE_INPUT_RECORDS over-counts the rows an input will deliver

### PR DESCRIPTION
**Pull request by deniskuzZ**
- An announcement carries the source task's output record count so far, not a per-event delta; keep the latest value per input instead of summing events, so a pipelined input's earlier spills are not counted again
- Count an input that wrote no rows in the denominator; leaving it out while numInputs keeps it in the multiplier spread the mean of the inputs that had data over the ones that did not
- Announce from the composite event path too: which path runs depends only on tez.am.shuffle.auxiliary-service.id, so on the Tez shuffle handler the counter was never updated at all
- Multiply before dividing, the mean was truncated first
- Saturate the announced count instead of wrapping it through int, and include large records, which bypass OUTPUT_RECORDS -- the VertexManager event already counts both
- updateApproximateInputRecords takes the input index
- Measured before the fix: 10 inputs, 2 announcing 1000 rows each, read 10,000 instead of 2,000; 4 pipelined events totalling 600 read 450

### CODE REVIEW

**Comment by tez-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 12s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 50s |  |  master passed  |
| +1 :green_heart: |  compile  |   4m  0s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 36s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  master passed  |
| +0 :ok: |  spotbugs  |   1m 14s |  |  tez-runtime-library in master has 143 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 40s |  |  the patch passed  |
| +1 :green_heart: |  codespell  |   1m 34s |  |  No new issues.  |
| +1 :green_heart: |  compile  |   4m  4s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m  4s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 33s | [/results-checkstyle-tez-runtime-library.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/1/artifact/out/results-checkstyle-tez-runtime-library.txt) |  tez-runtime-library: The patch generated 1 new + 77 unchanged - 1 fixed = 78 total (was 78)  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 28s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  74m  0s |  |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 32s |  |  The patch does not generate ASF License warnings.  |
|  |   |  99m 39s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.56 ServerAPI=1.56 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/1/artifact/out/Dockerfile |
| Optional Tests | dupname compile unit asflicense javac javadoc spotbugs checkstyle codespell detsecrets |
| uname | Linux c0c85768ffd3 5.15.0-190-generic #200-Ubuntu SMP Fri Aug 7 15:06:04 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | tez-personality.sh |
| git revision | master / 0e9f756609fe06835cb8f50e0f8e90a3e8282a6f |
| Default Java | Eclipse Adoptium-21.0.12+8-LTS |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/1/testReport/ |
| Max. process+thread count | 1405 (vs. ulimit of 5500) |
| modules | C: tez-runtime-library U: tez-runtime-library |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/1/console |
| versions | git=2.43.0 maven=3.9.15 spotbugs=4.9.3 codespell=2.4.1 |
| Powered by | Apache Yetus 0.15.1 https://yetus.apache.org |


This message was automatically generated.

File: tez-runtime-library/src/test/java/org/apache/tez/runtime/library/common/shuffle/impl/TestShuffleManager.java [hunk 1]
```diff
@@ -160,6 +161,77 @@ public void testMultiplePartitions() throws Exception {
         shuffleManager.getNumOfCompletedInputs());
   }
 
+  private long approximateInputRecords(InputContext inputContext) {
+    return inputContext.getCounters().findCounter(TaskCounter.APPROXIMATE_INPUT_RECORDS).getValue();
+  }
```
Comment (abstractdog) [inline tez-runtime-library/src/test/java/org/apache/tez/runtime/library/common/shuffle/impl/TestShuffleManager.java]: this method can go below the `@Test` methods

File: tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/shuffle/impl/ShuffleManager.java [hunk 2]
```diff
@@ -348,13 +352,27 @@ public ShuffleManager(InputContext inputContext, Configuration conf, int numInpu
...
+  /**
+   * Takes the row count one input announces and extrapolates the total over every input.
+   * An announcement is the source task's output record count so far, not a per-event delta, so a
+   * pipelined input announces a new total per spill and only its latest one is kept -- it is the
+   * count at the time the event was built, which the final event brings up to date. An input that
+   * wrote no rows announces zero and still joins the denominator, which numInputs already counts
+   * in the multiplier. A count the writer could not fit in an int arrives negative; ignore it
+   * rather than subtract it from the total.
+   */
+  public void updateApproximateInputRecords(int inputIndex, int numRecords) {
```
Comment (abstractdog) [inline tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/shuffle/impl/ShuffleManager.java]: nit: this is kind of shuffle internals, a package-protected scope makes more sense to me
I can seen it used to be public, but as the signature is already touched by this PR, maybe make it happen now

File: tez-runtime-library/src/test/java/org/apache/tez/runtime/library/common/shuffle/impl/TestShuffleManager.java
Comment (abstractdog) [inline tez-runtime-library/src/test/java/org/apache/tez/runtime/library/common/shuffle/impl/TestShuffleManager.java:1]: consider adding a unit test case for negative `numRecords`

**Review by abstractdog — CHANGES_REQUESTED** [review]
nice catch, left minor comments only
check also copilot comments please, if they make sense

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  8s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 3 new or modified test files.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  master passed  |
| +1 :green_heart: |  compile  |   2m 34s |  |  master passed  |
| -1 :x: |  bufcompat  |   0m  1s | [/branch-bufcompat-stderr.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/4/artifact/out/branch-bufcompat-stderr.txt) |  Error running buf. Please check buf stderr files.  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 24s |  |  master passed  |
| +0 :ok: |  spotbugs  |   0m 40s |  |  tez-runtime-library in master has 143 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 14s |  |  the patch passed  |
| -1 :x: |  codespell  |   0m 52s | [/results-codespell.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/4/artifact/out/results-codespell.txt) |  The patch generated 1 new + 10 unchanged - 0 fixed = 11 total (was 10)  |
| +1 :green_heart: |  compile  |   2m 30s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 30s |  |  the patch passed  |
| -1 :x: |  javac  |   2m 30s | [/results-compile-javac-root.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/4/artifact/out/results-compile-javac-root.txt) |  root generated 2 new + 706 unchanged - 0 fixed = 708 total (was 706)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  buflint  |   0m  5s |  |  No new issues.  |
| -0 :warning: |  checkstyle  |   0m 18s | [/results-checkstyle-tez-runtime-library.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/4/artifact/out/results-checkstyle-tez-runtime-library.txt) |  tez-runtime-library: The patch generated 3 new + 183 unchanged - 1 fixed = 186 total (was 184)  |
| +1 :green_heart: |  javadoc  |   0m 19s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 47s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  59m 41s |  |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 23s |  |  The patch does not generate ASF License warnings.  |
|  |   |  75m 24s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.56 ServerAPI=1.56 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/4/artifact/out/Dockerfile |
| Optional Tests | dupname compile unit asflicense javac javadoc spotbugs checkstyle codespell detsecrets cc buflint bufcompat |
| uname | Linux 7793c0caf8ad 5.15.0-190-generic #200-Ubuntu SMP Fri Aug 7 15:06:04 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | tez-personality.sh |
| git revision | master / 0e9f756609fe06835cb8f50e0f8e90a3e8282a6f |
| Default Java | Eclipse Adoptium-21.0.12+8-LTS |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/4/testReport/ |
| Max. process+thread count | 1417 (vs. ulimit of 5500) |
| modules | C: tez-runtime-library U: tez-runtime-library |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/4/console |
| versions | git=2.43.0 maven=3.9.15 buf=1.68.2 spotbugs=4.9.3 codespell=2.4.1 |
| Powered by | Apache Yetus 0.15.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 12s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 4 new or modified test files.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  4s |  |  master passed  |
| +1 :green_heart: |  compile  |   4m  4s |  |  master passed  |
| -1 :x: |  bufcompat  |   0m  2s | [/branch-bufcompat-stderr.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/8/artifact/out/branch-bufcompat-stderr.txt) |  Error running buf. Please check buf stderr files.  |
| +1 :green_heart: |  checkstyle  |   0m 41s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  2s |  |  master passed  |
| +0 :ok: |  spotbugs  |   1m 18s |  |  tez-runtime-library in master has 143 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 43s |  |  the patch passed  |
| -1 :x: |  codespell  |   1m 34s | [/results-codespell.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/8/artifact/out/results-codespell.txt) |  The patch generated 1 new + 13 unchanged - 0 fixed = 14 total (was 13)  |
| +1 :green_heart: |  compile  |   4m  5s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   4m  5s |  |  the patch passed  |
| -1 :x: |  javac  |   4m  5s | [/results-compile-javac-root.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/8/artifact/out/results-compile-javac-root.txt) |  root generated 2 new + 706 unchanged - 0 fixed = 708 total (was 706)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  buflint  |   0m  9s |  |  No new issues.  |
| -0 :warning: |  checkstyle  |   0m 34s | [/results-checkstyle-tez-runtime-library.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/8/artifact/out/results-checkstyle-tez-runtime-library.txt) |  tez-runtime-library: The patch generated 6 new + 217 unchanged - 1 fixed = 223 total (was 218)  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 26s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  75m 42s |  |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 102m 24s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.56 ServerAPI=1.56 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/8/artifact/out/Dockerfile |
| Optional Tests | dupname compile unit asflicense javac javadoc spotbugs checkstyle codespell detsecrets cc buflint bufcompat |
| uname | Linux 9235b2c4fc18 5.15.0-190-generic #200-Ubuntu SMP Fri Aug 7 15:06:04 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | tez-personality.sh |
| git revision | master / 0e9f756609fe06835cb8f50e0f8e90a3e8282a6f |
| Default Java | Eclipse Adoptium-21.0.12+8-LTS |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/8/testReport/ |
| Max. process+thread count | 2124 (vs. ulimit of 5500) |
| modules | C: tez-runtime-library U: tez-runtime-library |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/8/console |
| versions | git=2.43.0 maven=3.9.15 buf=1.68.2 spotbugs=4.9.3 codespell=2.4.1 |
| Powered by | Apache Yetus 0.15.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 13s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 4 new or modified test files.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 50s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 39s |  |  master passed  |
| +1 :green_heart: |  compile  |   4m  1s |  |  master passed  |
| -1 :x: |  bufcompat  |   0m  2s | [/branch-bufcompat-stderr.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/9/artifact/out/branch-bufcompat-stderr.txt) |  Error running buf. Please check buf stderr files.  |
| +1 :green_heart: |  checkstyle  |   1m 39s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m  9s |  |  master passed  |
| +0 :ok: |  spotbugs  |   1m 15s |  |  tez-runtime-library in master has 143 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   6m 37s |  |  root in master has 1054 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 53s |  |  the patch passed  |
| +1 :green_heart: |  codespell  |   1m 34s |  |  The patch generated 0 new + 5 unchanged - 8 fixed = 5 total (was 13)  |
| +1 :green_heart: |  compile  |   4m  2s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   4m  2s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m  2s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  buflint  |   0m  9s |  |  No new issues.  |
| -0 :warning: |  checkstyle  |   0m 32s | [/results-checkstyle-tez-runtime-library.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/9/artifact/out/results-checkstyle-tez-runtime-library.txt) |  tez-runtime-library: The patch generated 6 new + 217 unchanged - 1 fixed = 223 total (was 218)  |
| -0 :warning: |  checkstyle  |   1m  0s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/9/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 6 new + 217 unchanged - 1 fixed = 223 total (was 218)  |
| +1 :green_heart: |  javadoc  |   2m  4s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   8m  9s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  75m 39s |  |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 22s |  |  The patch does not generate ASF License warnings.  |
|  |   | 122m 50s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.56 ServerAPI=1.56 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/9/artifact/out/Dockerfile |
| Optional Tests | dupname compile unit asflicense codespell detsecrets javac javadoc spotbugs checkstyle cc buflint bufcompat |
| uname | Linux c22f869cb033 5.15.0-190-generic #200-Ubuntu SMP Fri Aug 7 15:06:04 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | tez-personality.sh |
| git revision | master / 0e9f756609fe06835cb8f50e0f8e90a3e8282a6f |
| Default Java | Eclipse Adoptium-21.0.12+8-LTS |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/9/testReport/ |
| Max. process+thread count | 1372 (vs. ulimit of 5500) |
| modules | C: tez-runtime-library . U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/9/console |
| versions | git=2.43.0 maven=3.9.15 buf=1.68.2 spotbugs=4.9.3 codespell=2.4.1 |
| Powered by | Apache Yetus 0.15.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 15s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 4 new or modified test files.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 53s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  5s |  |  master passed  |
| +1 :green_heart: |  compile  |   4m  4s |  |  master passed  |
| -1 :x: |  bufcompat  |   0m  2s | [/branch-bufcompat-stderr.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/10/artifact/out/branch-bufcompat-stderr.txt) |  Error running buf. Please check buf stderr files.  |
| +1 :green_heart: |  checkstyle  |   1m 46s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m  8s |  |  master passed  |
| +0 :ok: |  spotbugs  |   1m 18s |  |  tez-runtime-library in master has 143 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   6m 32s |  |  root in master has 1054 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 25s |  |  the patch passed  |
| +1 :green_heart: |  codespell  |   1m 52s |  |  The patch generated 0 new + 5 unchanged - 8 fixed = 5 total (was 13)  |
| +1 :green_heart: |  compile  |   4m 15s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   4m 15s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 15s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  buflint  |   0m  9s |  |  No new issues.  |
| -0 :warning: |  checkstyle  |   0m 30s | [/results-checkstyle-tez-runtime-library.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/10/artifact/out/results-checkstyle-tez-runtime-library.txt) |  tez-runtime-library: The patch generated 6 new + 217 unchanged - 1 fixed = 223 total (was 218)  |
| -0 :warning: |  checkstyle  |   1m  2s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/10/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 6 new + 217 unchanged - 1 fixed = 223 total (was 218)  |
| +1 :green_heart: |  javadoc  |   2m  7s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   8m 34s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  75m 19s |  |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 24s |  |  The patch does not generate ASF License warnings.  |
|  |   | 124m 51s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.56 ServerAPI=1.56 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/10/artifact/out/Dockerfile |
| Optional Tests | dupname compile unit asflicense codespell detsecrets javac javadoc spotbugs checkstyle cc buflint bufcompat |
| uname | Linux 5d6e8734283d 5.15.0-190-generic #200-Ubuntu SMP Fri Aug 7 15:06:04 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | tez-personality.sh |
| git revision | master / 0e9f756609fe06835cb8f50e0f8e90a3e8282a6f |
| Default Java | Eclipse Adoptium-21.0.12+8-LTS |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/10/testReport/ |
| Max. process+thread count | 1544 (vs. ulimit of 5500) |
| modules | C: tez-runtime-library . U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-540/10/console |
| versions | git=2.43.0 maven=3.9.15 buf=1.68.2 spotbugs=4.9.3 codespell=2.4.1 |
| Powered by | Apache Yetus 0.15.1 https://yetus.apache.org |


This message was automatically generated.
